utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -484,104 +654,199 @@ impl<'a, 'b> SyscallsInvocator<'a, 'b> {
-                            Instruction::I32Const(value_set_ptr),
-                            Instruction::Call(gr_source_call_indexes_handle),
-                        ];
-                        // Generate value definition instructions.
-                        // Value data is put right after `gr_source` bytes (value_set_ptr + hash len).
-                        let mut value_instr = utils::translate_ptr_data(
-                            WasmWords::new(self.unstructured.int_in_range(range)?.to_le_bytes()),
-                            (value_set_ptr + mem::size_of::<Hash>() as i32, value_set_ptr),
-                        );
-                        ret_instr.append(&mut value_instr);
-
-                        ret_instr
-                    }
-                    ActorKind::ExistingAddresses(addresses) => {
-                        let address_words = WasmWords::new(
-                            *self
-                                .unstructured
-                                .choose(&utils::non_empty_to_vec(addresses))?,
-                        );
-                        let value_words =
-                            WasmWords::new(self.unstructured.int_in_range(range)?.to_le_bytes());
-                        utils::translate_ptr_data(
-                            address_words.merge(value_words),
-                            (value_set_ptr, value_set_ptr),
-                        )
-                    }
-                    ActorKind::Random => {
-                        let random_address_words =
-                            WasmWords::new(self.unstructured.arbitrary::<[u8; 32]>()?);
-                        let value_words =
-                            WasmWords::new(self.unstructured.int_in_range(range)?.to_le_bytes());
-                        utils::translate_ptr_data(
-                            random_address_words.merge(value_words),
-                            (value_set_ptr, value_set_ptr),
-                        )
-                    }
+                let mut ret_instr =
+                    self.build_actor_id_instructions(actor, (value_set_ptr, None))?;
+
+                // Generate value definition instructions.
+                // Value data is put right after actor id bytes (value_set_ptr + hash len).
+                let mut value_instr = utils::translate_ptr_data(
+                    WasmWords::new(self.unstructured.int_in_range(range)?.to_le_bytes()),
+                    (
+                        value_set_ptr + mem::size_of::<Hash>() as i32,
+                        Some(value_set_ptr),
+                    ),
+                );
+                ret_instr.append(&mut value_instr);
+
+                ret_instr
+            }
+            ref ptr_allowed_values @ (PtrParamAllowedValues::ReservationIdWithValue(_)
+            | PtrParamAllowedValues::ReservationIdWithActorIdAndValue {
+                ..
+            }
+            | PtrParamAllowedValues::ReservationId) => {
```

Comment:
these must be tested with unit tests in this PR

Reply:
possibly problematic due to current test architecture

Reply:
discussed, introduce in a new PR

Reply:
done in 35ace4bdcb10a44460f457b09eb8e524953b8aa7 without refactoring `execute_wasm_with_custom_configs` arguments